elements/Shape.js

```diff
@@ -10,6 +11,17 @@ class Shape extends Component {
         });
         this.state = this.touchableGetInitialState();
     }
+
+    setNativeProps = (args) => {
+        var extracted = extractProps(args, this)
```

Comment:
Thanks for your comment.  The for loop is just limiting the values that get changed to the ones returned by extractProps().  

So if extractProps() does what you want, then that value will persist.  Anything else is left alone. It looks like extractProps() does support transform matrices.  So I think that any transform you use without setNativeProps should also keep working here, no?